src/XMakeBuildEngine/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -490,8 +490,18 @@ private int LaunchNode(string msbuildLocation, string commandLineArgs)
             CommunicationsUtilities.Trace("Launching node from {0}", msbuildLocation);
 
 #if RUNTIME_TYPE_NETCORE
-            string coreRunName = NativeMethodsShared.IsUnixLike ? "corerun" : "CoreRun.exe";
-            string exeName = Path.Combine(Path.GetDirectoryName(msbuildLocation), coreRunName);
+            // Try to get the process name in case of different host running.
+            string hostName, exeName;
+            using (Process currentProcess = Process.GetCurrentProcess()) hostName = currentProcess.MainModule.FileName;
+            if (string.IsNullOrEmpty(hostName))
+            {
+                hostName = NativeMethodsShared.IsUnixLike ? "corerun" : "CoreRun.exe";
```

Comment:
I kind of prefer to not have corerun hard-coded at all and if we end up finding cases then we should come up with a better fix.